src/main/java/be/seeseemelk/mockbukkit/block/state/BlockStateMock.java

```diff
@@ -274,14 +298,10 @@ public boolean equals(Object obj)
 		{
 			return false;
 		}
-		if (this.isPlaced() && this.getLocation() != other.getLocation() && (this.getLocation() == null || !this.getLocation().equals(other.getLocation())))
-		{
-			return false;
-		}
+		return !this.isPlaced() || this.getLocation() == other.getLocation() || (this.getLocation() != null && this.getLocation().equals(other.getLocation()));
```

Comment:
Just a note in case anyone wants to bother checking this:
I was too lazy to check the equivalency of this statement 🤷🏻

Reply:
[Here's the CraftBukkit implementation](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/block/CraftBlockState.java#264-283). The only real difference is we skip the check if it isn't placed to avoid an exception being thrown when getting the location, which doesn't happen in CraftBukkit.

Reply:
I know, I was just commenting on the fact that I didn't check if the statement is equivalent to the one before since you basically inverted it. It was still an approving review, just a comment in case someone comes back here

Reply:
So can this be resolved? It's blocking the merge 😅

Reply:
of course